kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -1,53 +1,123 @@
-    const subscriptionFormRef = useRef(null);
+    const userData = useContext(UserProvider.context);
+
     const [isSubmitting, setIsSubmitting] = useState(false);
     const [paymentAuthorized, setPaymentAuthorized] = useState(false);
 
-    const toggleButton = () => {
-        setPaymentAuthorized(!paymentAuthorized);
-    };
+    const token = useRef(null);
 
-    /**
-     * Opens Stripe modal allowing a user to complete their subscription.
-     * @param {Object} event - The form submit event
-     */
-    const submit = event => {
+    const stripeLoadingPromise = useConditionalScriptLoad(
+        'https://checkout.stripe.com/checkout.js',
+        paymentAuthorized
+    );
+
+    function togglePaymentAuthorized() {
+        const newValue = !paymentAuthorized;
+        setPaymentAuthorized(newValue);
+    }
+
+    function openStripeModal(event) {
         event.preventDefault();
 
-        const subscriptionForm = subscriptionFormRef.current;
-        if (subscriptionForm) {
-            const formData = new FormData(subscriptionForm);
-            setIsSubmitting(true);
+        setIsSubmitting(true);
+
+        if (!stripeLoadingPromise) {
+            console.error('stripe script load was not started');
+            return;
+        }
 
+        stripeLoadingPromise.then(() => {
             const stripeHandler = window.StripeCheckout.configure({
                 key: window.mdn.stripePublicKey,
-                locale: 'auto',
+                locale,
                 name: 'MDN Web Docs',
                 zipCode: true,
                 currency: 'usd',
                 amount: 500,
-                email: '',
-                token: function(response) {
-                    formData.set('stripe_token', response.id);
-                    subscriptionForm.submit();
+                email: userData ? userData.email : '',
+                token(response) {
+                    token.current = response.id;
+                    createSubscription();
                 },
-                closed: function() {
-                    if (!formData.get('stripe_token')) {
+                closed() {
+                    if (!token.current) {
                         setIsSubmitting(false);
                     }
                 }
             });
-
             stripeHandler.open();
-        }
-    };
+        });
+    }
+
+    function createSubscription() {
+        fetch(SUBSCRIPTION_URL, {
+            method: 'POST',
+            body: JSON.stringify({
+                stripe_token: token.current // eslint-disable-line camelcase
+            }),
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken'),
+                'Content-Type': 'application/json'
+            }
+        })
+            .then(() => {
+                window.location = `/${locale}/payments/thank-you/`;
+            })
+            .catch(e => {
+                console.error('error while creating subscription', e);
+                alert(
```

Comment:
I'd much rather you use `setSubscriptionResponseError(e)` or something. And then, instead of `alert()` use...
```javascript

if (subscriptionResponseError) {
  return <p className="subscription-response-error">
    {                    gettext(
                        "An error occurred trying to set up the subscription with Stripe's server. We've recorded the error and will investigate it."
                    )
}
  </p>
}

return <form>
...

```
(sorry about the formatting)